tests/util.py

```diff
@@ -67,5 +71,43 @@ def mock_request(params: dict = None, data=None, **headers) -> Request:
         environ = create_environ(base_url='http://localhost:5000/', data=data)
     environ.update(headers)
     request = Request(environ)
-    request.args = ImmutableMultiDict(params.items())
+    request.args = ImmutableMultiDict(params.items())  # noqa
     return request
+
+
+@contextmanager
+def mock_client(config_file: str = 'pygeoapi-test-config.yml',
+                use_cookies: bool = True, **kwargs) -> FlaskClient:
```

Comment:
Is it reasonable to have the same mock client and test for the other supported frameworks?

Reply:
I thought it was too much work for now (I'm on a budget...) so I focused on the default one. Tbh, I was actually disappointed that no one ever bothered to add mock clients for any framework. So I'd say this already is a win 😉

Reply:
Thinking.. perhaps I misunderstood you. You mean use the mock FlaskClient to also test Django and Starlette? For Starlette it might work (since that also uses Werkzeug, from which the FlaskClient inherits its mock client), for Django I'm not so sure.
We would also have to make any supported framework a dependency/requirement for the test and install that framework prior to running the tests.

Reply:
Well, actually I mean separate StarletteClient and DjangoClient

Reply:
Ah okay, so I did understand you correctly the first time 🙂 
I can see what I can do in the time available, focusing on Starlette first. 
Are these mock clients for all frameworks a "deal breaker" for you?

Reply:
Ok, I'll have a look later this week!